### DISCUSSION TITLE
Are shifter proxies Multi-VO ready ?

### DISCUSSION THREAD

**Discussion by martynia** [DIRACGrid/DIRAC#5678] [Q&A]
Hello,

I would like to upload pilot log files to a Dirac  SE.  I looked into the docs and enabled shifter proxy by calling:

```python
self.am_setOption('shifterProxy', 'DataManager')
```
This call looks suspicious, since there is no VO in it, so how would Dirac magically know which proxy to use ? Mine is defined for gridpp VO, so I'm getting, no surprise:

`ERROR: Failed to set shifter proxy No shifter User defined for DataManager`

Looking at the code it becomes clear why, the Operation() are instantiated w/o the VO:

[https://github.com/DIRACGrid/DIRAC/blob/d6449a3d6821780b613c174b3c7026c61a8cf6d2/src/DIRAC/Core/Utilities/Shifter.py#L30
](https://github.com/DIRACGrid/DIRAC/blob/d6449a3d6821780b613c174b3c7026c61a8cf6d2/src/DIRAC/Core/Utilities/Shifter.py#L30)

In `PilotSynchAgent`, there is an upload attempt to an SE w/o a shifter proxy apparently (no `am_setOption` call),
but this would only work if the host certificate joins a VO, wouldn't it ?

Any suggestions ?

cheers, JM

**Comment by fstagni** [comment]
`PilotSyncAgent` would need a shifter proxy if it is uploading a Grid SE. Indeed this is something that I am not sure it has ever been tried, and indeed it might not work ATM.

Anyway, in the general case it seems that indeed setting a shifter proxy is not multi-VO ready. Do you want to give it a try in coding a solution?

**Comment by martynia** [comment]
Thanks for the reply. I will look into it and try, hopefully not breaking a single VO system.
More to come.

Thanks, JM

**Comment by martynia** [comment]
Hi,

I think I have a working proposal for this. Which release version should I make a PR against ?
Thanks, JM

**Reply by andresailer** [reply]
It depends how intrusive the changes are in the end, but given the potential of disruption for these changes, probably integration.
